backend/src/models/users.go

```diff
@@ -56,7 +55,7 @@ func (User) TableName() string {
 
 func (usr *User) BeforeCreate(tx *gorm.DB) error {
 	if usr.Email == "" {
-		usr.Email = usr.NameLast + "." + usr.NameFirst + "@unlocked.v2"
+		usr.Email = usr.Username + "@unlocked.v2"
```

Comment:
right now we have to use the `preferred_username` field of the oauth id_token (which is what is sent to the client/canvas to identify the user), because the `sub` field (default) will be set to the users `kratos_id` instead of the username.

It's possible we may have to either use our `{username}@unlocked.v2` as the de-facto external identifier for our users in providers, or duplicate the username field in the email? or use the users kratos_id (uuid) as the SIS id and log them in through their SIS id? not sure yet